### ISSUE TITLE
MaxPool should be corrected and documented properly

### ISSUE THREAD

**Issue by JagdishKolhe** [onnx/onnx#5711]
# Bug Report

### Is the issue related to model conversion?
Yes and No

### Describe the bug
When I run the single node model with kernel_size 1*1, the ceil_mode docent make any sense. The output shape should always be floor.

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 20.04*): Linux Ubuntu 20.04
- ONNX version (*e.g. 1.13*): 1.13
- Python version: 3.10

### Reproduction instructions

```
import torch
import torch.nn as nn
from torch.nn.parameter import Parameter

class TorchNetwork(nn.Module):
    def __init__(self):
        super(TorchNetwork, self).__init__()
        self.maxpool_1 = nn.MaxPool2d(kernel_size = [1, 1], stride = [2, 2], padding = [0, 0],ceil_mode=True)

    def forward(self, V_0):
        V_maxpool = self.maxpool_1(V_0)
        return V_maxpool

input_tensor = torch.randn(4, 4, 12, 12)
```

When I run above model in torch 2.0 with input as input_tensor, it always gives output shape as (4, 4, 6, 6) ignoring ceil_mode as kernel_size is 1*1 and padding is 0, which is expected behavior.
Whereas, when I export above model in onnx and run it I get output shape as (4, 4, 7, 7) which should not be the case. Although its the converter issue, still this validation should happen in ONNX runtime and runtime should through error.

### Expected behavior
ONNX Runtime should validate input and output shape w.r.t kernel_size, padding and dilation and throw the error.

**Comment by liqunfu** [comment]
It is true this is a bug in converter.  I think ONNX shall have test cases to cover this and Torch exporter and ORT shall implement the correct behavior.

**Comment by titaiwangms** [comment]
Hi @liqunfu @gramalingam ,
https://github.com/pytorch/pytorch/issues/113155

This is the same issue as https://github.com/microsoft/onnxruntime/issues/16203. Is there anyway we can make a change on ORT side?

**Comment by titaiwangms** [comment]
Should be fixed by https://github.com/pytorch/pytorch/pull/106270

**Comment by titaiwangms** [comment]
Also relateed to https://github.com/microsoft/onnxruntime/issues/15446
